wagtail/admin/views/workflows.py

```diff
@@ -393,24 +399,86 @@ def delete_action(self):
         self.object.deactivate(user=self.request.user)
 
 
-def usage(request, pk):
-    workflow = get_object_or_404(Workflow, id=pk)
+class WorkflowPageFilterSet(PageIndexView.filterset_class):
+    content_type = PageTypeFilter()
 
-    editable_pages = page_permission_policy.instances_user_has_permission_for(
-        request.user, "change"
-    )
-    pages = workflow.all_pages() & editable_pages
-    paginator = Paginator(pages, per_page=10)
-    pages = paginator.get_page(request.GET.get("p"))
 
-    return render(
-        request,
-        "wagtailadmin/workflows/usage.html",
-        {
-            "workflow": workflow,
-            "used_by": pages,
-        },
-    )
+class WorkflowUsageView(PageIndexView):
+    permission_policy = page_permission_policy
+    pk_url_kwarg = "pk"
+    index_url_name = "wagtailadmin_workflows:usage"
+    index_results_url_name = "wagtailadmin_workflows:usage_results"
+    header_icon = "tasks"
+    filterset_class = WorkflowPageFilterSet
+    page_title = _("Usage")
+    paginate_by = 20
+    # Avoid ambiguity of whether this view is working with the page or the workflow,
+    # and let the view code be more explicit about which model is being worked with.
+    model = None
```

Comment:
Setting this to `Workflow` would work, and will be consistent with what `self.object` is, but it's a bit confusing since the view lists page objects.

On other `UsageView`s that are powered by the `ReferenceIndex` model, it's somewhat acceptable since the listed items are of variable models, and you still work with the "used" object's permission policy.

Setting this to `Page` wouldn't work because `LocaleMixin.get_locale()` (that is used by the `generic.IndexView`) assumes `self.object` is an instance of `self.model`. It would crash because `Page` (`self.model`) is translatable, but `Workflow` (`self.object`) is not.

While we could refactor the mixin to be more defensive, I think in this case we're better off not setting the model and just explicitly refer to the model where we need to.